graph-builder/src/graph.rs

```diff
@@ -23,12 +23,16 @@ use failure::{Error, Fallible};
 use futures::Future;
 use lazy_static;
 pub use parking_lot::RwLock;
-use prometheus::{self, Counter, IntGauge};
+use prometheus::{self, histogram_opts, Counter, Gauge, Histogram, IntGauge};
 use serde_json;
 use std::collections::{HashMap, HashSet};
 use std::sync::Arc;
 use std::thread;
 
+/// Custom bucket values for upstream scraping duration in seconds
+pub const UPSTREAM_SCRAPE_BUCKETS: &[f64; 11] =
+    &[1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0];
```

Comment:
Do you mind explaining how you chose these values and why it is eleven of them rather than some other arbitrary value?

Reply:
Since this is entirely new metric all values (and the number of them) is arbitrary - I ran `curl --verbose --header 'Accept:application/json' http://localhost:9080/metrics` using test quay repo and values landed in 6-7 buckets usually, thus 11 values from 1.0 to 11.0

It may be entirely different on stage/prod deployments, so soon these would be updated

Reply:
I'm starting to grasp it. Do we want the first scrape to land within one of these buckets too?

Reply:
Self answer: no, because we don't even measure it.